ui/src/app/views/resource-card/component.tsx

```diff
@@ -113,175 +113,310 @@ export class ResourceCardMenuComponent extends ReactWrapperBase {
+  cardName: {
+    fontSize: '18px',
+    fontWeight: 500,
+    lineHeight: '22px',
+    color: '#2691D0',
+    cursor: 'pointer',
+    wordBreak: 'break-all'
+  },
+  lastModified: {
+    color: '#4A4A4A',
+    fontSize: '11px',
+    display: 'inline-block',
+    lineHeight: '14px',
+    fontWeight: 300,
+    marginBottom: '0.2rem'
+  },
+  resourceType: {
+    height: '22px',
+    width: 'max-content',
+    paddingLeft: '10px',
+    paddingRight: '10px',
+    borderRadius: '4px 4px 0 0',
+    display: 'flex',
+    justifyContent: 'center',
+    color: '#FFFFFF',
+    fontFamily: 'Montserrat, sans-serif',
+    fontSize: '12px',
+    fontWeight: 500
+  },
+  cardFooter: {
+    display: 'flex',
+    flexDirection: 'column'
+  }
+});
+
+const resourceTypeStyles = reactStyles({
+  cohort: {
+    backgroundColor: '#F8C954'
+  },
+  conceptSet: {
+    backgroundColor: '#AB87B3'
+  },
+  notebook: {
+    backgroundColor: '#8BC990'
+  }
+});
+
+export interface ResourceCardProps {
   resourceCard: RecentResource;
-  @Input('cssClass')
-  cssClass: string;
-  @Output() onUpdate: EventEmitter<void | NotebookRename> = new EventEmitter();
-  wsNamespace: string;
-  wsId: string;
-  resource: any;
-  router: Router;
-  invalidResourceError = false;
-  confirmDeleting = false;
-
-  renaming = false;
-  editing = false;
-
-  constructor(
-    private cohortsService: CohortsService,
-    private workspacesService: WorkspacesService,
-    private conceptSetsService: ConceptSetsService,
-    private signInService: SignInService,
-    private route: Router,
-  ) {}
-
-  ngOnInit() {
-    this.wsNamespace = this.resourceCard.workspaceNamespace;
-    // RW-1298 this should be updated
-    this.wsId = this.resourceCard.workspaceFirecloudName;
-    if (this.resourceCard) {
-      if (this.resourceCard.notebook) {
-        this.resourceType = ResourceType.NOTEBOOK;
-      } else if (this.resourceCard.cohort) {
-        this.resourceType = ResourceType.COHORT;
-      } else if (this.resourceCard.conceptSet) {
-        this.resourceType = ResourceType.CONCEPT_SET;
+  onUpdate: Function;
+}
+
+export interface ResourceCardState {
+  renaming: boolean;
+  editing: boolean;
+  confirmDeleting: boolean;
+  invalidResourceError: boolean;
+}
+
+export class ResourceCard extends React.Component<ResourceCardProps, ResourceCardState> {
+
+  constructor(props: ResourceCardProps) {
+    super(props);
+    const defaultState = {
+      editing: false,
+      renaming: false,
+      confirmDeleting: false
+    };
+    if (props.resourceCard) {
+      if ((props.resourceCard.notebook) ||
+          (props.resourceCard.cohort) ||
+          (props.resourceCard.conceptSet)) {
+        this.state = {
+          invalidResourceError: false,
+          ...defaultState
+        };
       } else {
-        this.resourceType = ResourceType.INVALID;
-        this.invalidResourceError = true;
+        this.state = {
+          invalidResourceError: true,
+          ...defaultState
+        };
```

Comment:
I personally don't really see how an "invalid resource" could ever come about, since we control the creation of resource cards which stem from creating cohorts, concept sets, and notebooks only.  maybe @blrubenstein or @calbach have thoughts on the initial implementation?

Reply:
Yes, it indicates a programming error and as such we probably shouldn't be driving UI with it. My suggestion would be that this component treats "oneof notebook/cohort/conceptSet" as a precondition, and that the parent component does any necessary validation on the list (logs errors and filters out unsupported types).

I suspect this originated when we added "concept sets" to the recent resources and subsequently broke the resource cards in an unpleasant way. So I think validating this *somewhere* would be good.

Reply:
Not addressed (or maybe it was?). Please resolve these kinds of comment threads before merging ideally by replying